packages/phone-number-privacy/combiner/test/integration/domain.test.ts

```diff
@@ -617,492 +599,683 @@ describe('domainService', () => {
+        new Map([
+          [
+            `${DefaultKeyName.DOMAINS}-1`,
+            '02000000b03e8d5203edb8b27a9c56185df0ee94e8be45f0c91e116beac68c851cc4ba10',
+          ],
+        ])
+      )
+      keyProvider3 = new MockKeyProvider(
+        new Map([
+          [
+            `${DefaultKeyName.DOMAINS}-1`,
+            '03000000bad95bc039a5418bd57e7f5bad4bf6c19ff85e523462925e226e6ac0a6770005',
+          ],
+        ])
+      )
+      keyProvider4 = new MockKeyProvider(
+        new Map([
+          [
+            `${DefaultKeyName.DOMAINS}-1`,
+            '040000002ffdcc94fd5322e4f3e0d7ba00b591c38f77e584b61b59dd6e62cc6cfed5fd06',
+          ],
+        ])
+      )
+      keyProvider5 = new MockKeyProvider(
+        new Map([
+          [
+            `${DefaultKeyName.DOMAINS}-1`,
+            '05000000243505a19a546f5002511f9527fe03401908cd6427991f69b2380e355fec0d0d',
+          ],
+        ])
+      )
+      app = startCombiner(combinerConfigLargerN, getContractKit(combinerConfigLargerN.blockchain))
+    })
+
+    beforeEach(async () => {
+      signerDB1 = await initSignerDatabase(signerConfig, signerMigrationsPath)
+      signerDB2 = await initSignerDatabase(signerConfig, signerMigrationsPath)
+      signerDB3 = await initSignerDatabase(signerConfig, signerMigrationsPath)
+      signerDB4 = await initSignerDatabase(signerConfig, signerMigrationsPath)
+      signerDB5 = await initSignerDatabase(signerConfig, signerMigrationsPath)
+
+      signer1 = startSigner(signerConfig, signerDB1, keyProvider1).listen(3001)
+      signer2 = startSigner(signerConfig, signerDB2, keyProvider2).listen(3002)
+      signer3 = startSigner(signerConfig, signerDB3, keyProvider3).listen(3003)
+      signer4 = startSigner(signerConfig, signerDB4, keyProvider4).listen(3004)
+      signer5 = startSigner(signerConfig, signerDB5, keyProvider5).listen(3005)
+    })
+
+    afterEach(async () => {
+      await signerDB1?.destroy()
+      await signerDB2?.destroy()
+      await signerDB3?.destroy()
+      await signerDB4?.destroy()
+      await signerDB5?.destroy()
+      signer1?.close()
+      signer2?.close()
+      signer3?.close()
+      signer4?.close()
+      signer5?.close()
+    })
+
+    it('Should respond with 200 on valid request', async () => {
+      const [req, poprfClient] = await signatureRequest(undefined, undefined, DOMAINS_PUBKEY_N5_T4)
+      const res = await request(app).post(CombinerEndpoint.DOMAIN_SIGN).send(req)
+
+      expect(res.status).toBe(200)
+      expect(res.body).toStrictEqual<DomainRestrictedSignatureResponse>({
+        success: true,
+        version: res.body.version,
+        signature: res.body.signature,
+        status: {
+          disabled: false,
+          counter: 1,
+          timer: res.body.status.timer,
+          now: res.body.status.now,
+        },
+      })
+      poprfClient.unblindResponse(Buffer.from(res.body.signature, 'base64'))
+    })
+
+    it('Should respond with 429 on out of quota', async () => {
```

Comment:
So this was returning 502 before, right? Might be good to have a comment explaining that